railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
See: https://github.com/rails/rails/pull/24654/commits/5e96d79f8d06b883eb7b3fd77df9eef3431d2ad2

So two things now:
- http://take.ms/Mr4Qk is there some actionable on this?
- Is https://github.com/rails/rails/pull/24654/commits/5e96d79f8d06b883eb7b3fd77df9eef3431d2ad2 enough to handle engines?